docs/book/products/search.rst

```diff
@@ -0,0 +1,55 @@
+.. index::
+   single: Search
+
+Search
+======
+
+Having a products search functionality in an ecommerce system is a very popular usecase. Sylius provides a products search functionality
+that is a grid filter.
+
+Grid filters
+------------
+
+For simple usecases of products search use the **filters of grids**.
+The products grid in the shop has already a ``search`` filter:
+
+.. code-blok:: yaml
+
+    # Sylius/Bundle/ShopBundle/Resources/config/grids/product.yml
+    filters:
+        search:
+            type: string
+            label: false
+            options:
+                fields: [translation.name]
+            form_options:
+                type: contains
+
+It searches by product names that contain a string that the user typed in.
+
+Customizing search filter
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The search bar in many shops should be more sophisticated, than just a simple text search. You may be needing to add
+searching by price, reviews, sizes or colors.
+
+If you would like to extend this built-in functionality read
+:doc:`this article about grids customizations </customization/grids>`, and :doc:`the GridBundle docs </bundles/SyliusGridBundle/index>`.
+
+ElasticSearch
+-------------
+
+When the grids filtering is not enough for you, and your needs are more complex you should go for the
+`ElasticSearch <https://www.elastic.co/products/elasticsearch>`_ integration.
+
+Sylius is provided with the `Lakion/SyliusElasticSearchBundle <https://github.com/Lakion/SyliusElasticSearchBundle>`_ integration.
```

Comment:
`Sylius is provided with the` it is not totally true. `You can extend your Sylius with external plugin which will provide  ElasticSearch functionality`?